kroxylicious-filters/kroxylicious-simple-transform/src/main/java/io/kroxylicious/proxy/filter/simpletransform/ByteBufferTransformation.java

```diff
@@ -7,10 +7,12 @@
 
 import java.nio.ByteBuffer;
 
+import org.apache.kafka.common.Uuid;
+
 /**
  * A transformation of the key or value of a produce record.
  */
 @FunctionalInterface
 public interface ByteBufferTransformation {
-    ByteBuffer transform(String topicName, ByteBuffer original);
+    ByteBuffer transform(String topicName, ByteBuffer original, Uuid topicId);
```

Comment:
I know it is test code, but the API is odd.  At least put topicName and topicId together.  
Better, give responsibility to choose topicName or topicId to the call and leave this API as was.

Reply:
Actually, I expressed a different way to solve this down below.